src/resources/index.ts

```diff
@@ -19,33 +24,49 @@ export default async function(app: Express) {
       }
     });
   });
-  
+
 
   // TODO: build this
   app.get('/api/v1/resources/search', async (_, res) => {
       res.status(501);
       res.end(JSON.stringify('{"status":501}'));
   });
-  
+
   // TODO: build this
   app.get('/api/v1/resources/scan', async (_, res) => {
       res.status(501);
       res.end(JSON.stringify('{"status":501}'));
   });
-  
+
   /* Resource Life Cycle Routes */
 
-   /**
-   * Dummy route because GUI wants a content upload URL, even when we're just copying a YouTube link
-   * TO DO: remove the upload content requirement from the GUI, so that YouTube (or other web) links no longer have to post to an dummy file upload link
+  /*
+   * Endpoint for setting remote file metadata
    */
-    app.post('/api/v1/resources/:id/content/:token', authorizeRequest, async (req, res) => {
-      return res.status(200).send({
-        status: 200,
-        resource: {
-          id: req.params.id
-        }
-      })
+  app.post('/api/v1/resources/:id/content/:token', authorizeRequest, async (req, res) => {
+    let files = req.body?.remoteFiles || [];
+    files = files?.map((file: any) => {
+      if (!file?.mime && file?.mimeType) {
+        // copy mimetype
+        // The ayamel player requires mime to be set but
+        // the backend does not set a mime field.
+        file.mime = file.mimeType;
+      }
+      return file;
+    });
+
+    Resource.findByIdAndUpdate(req.params.id, { $set: { "content.files": files } }, { new: true }, (err, resource) => {
```

Comment:
Not sure if this is the best way to update docs in mongo. We are replacing the entire files array, I don't know if that will be an issue elsewhere.

Reply:
That should be fine for now. I think that was the behavior of the original resource library, but if it does cause problems we can try to figure out how to do a proper merge later.